SCClassLibrary/Common/Control/Server.sc

```diff
@@ -922,6 +946,7 @@ Server {
 		} {
 			this.disconnectSharedMemory;
 			pid = unixCmd(program ++ options.asOptionsString(addr.port), { |exitCode|
+				pid = nil;
 				this.prOnServerProcessExit(exitCode);
```

Comment:
shouldn't the line `pid = nil` happen in `prOnServerProcessExit` ?

Reply:
either is fine by me - had it there first - lets see if @brianlheim has a preference

Reply:
i wondered whether theoretically a unixCmd could return a non nil pid even after it has called the exit function.

Reply:
@telephon - do you mean, whether it's possible that unixCmd's callback could run while the process is still running? I looked over the associated code - I'm not an expert on the unixy stuff but it looks like it's a remote possibility? I think that would be an issue in the implementation of unixCmd though and IMO we're OK relying on it here.

Reply:
I agree with @telephon that the line `pid = nil` should go in `prOnServerProcessExit`, makes this easier to understand

Reply:
done